### PR TITLE
Update nodejs-extract-text-webpack to 3.0

### PR DESCRIPTION
**Pull request by dLobatog**

### CODE REVIEW

**Comment by mmoll** [comment]
it seems there are unrelated, old changes in this commit?

File: nodejs-extract-text-webpack-plugin/nodejs-bundle-extract-text-webpack-plugin.spec [hunk 1]
```diff
@@ -1,92 +1,85 @@
...
 Source15: http://registry.npmjs.org/co/-/co-4.6.0.tgz
 Source16: http://registry.npmjs.org/jsonify/-/jsonify-0.0.0.tgz
-Source17: extract-text-webpack-plugin-2.1.2-registry.npmjs.org.tgz
+Source17: extract-text-webpack-plugin-3.0.0-registry.npmjs.org.tgz
 BuildRequires: nodejs-packaging
-BuildRequires: npm
 BuildArch:  noarch
 ExclusiveArch: %{nodejs_arches} noarch
 
-Provides: npm(%{npm_name}) = %{version}
```
Comment (mmoll) [inline nodejs-extract-text-webpack-plugin/nodejs-bundle-extract-text-webpack-plugin.spec]: this line should probab;ly stay

**Comment by mmoll** [comment]
`nodejs-extract-text-webpack-plugin/extract-text-webpack-plugin-3.0.0-registry.npmjs.org.tgz` could get deleted and the spec adjusted, if needed, as the same file is there as annexed file.

File: nodejs-extract-text-webpack-plugin/nodejs-bundle-extract-text-webpack-plugin.spec (hunk 1 above)
Comment (dLobatog) [inline nodejs-extract-text-webpack-plugin/nodejs-bundle-extract-text-webpack-plugin.spec]: Afaik -`%{?nodejs_find_provides_and_requires}` does that

File: nodejs-extract-text-webpack-plugin/nodejs-bundle-extract-text-webpack-plugin.spec [hunk 2]
```diff
@@ -103,3 +96,5 @@ cp -pf CHANGELOG.md README.md ../../
 * Sat Jul 15 2017 Eric D. Helms <ericdhelms@gmail.com>
 - update webpack to v3.0 (ohadlevy@gmail.com)
 
+* Tue Jul 25 2017 author@example.com <Author Name> - 3.0.0-1
+- First release
```
Comment (mmoll) [inline nodejs-extract-text-webpack-plugin/nodejs-bundle-extract-text-webpack-plugin.spec]: aha? :)

**Comment by mmoll** [comment]
[test]

File: comps/comps-foreman-rhel7.xml [hunk 3]
```diff
@@ -258,26 +259,6 @@
...
-      <packagereq type="default">rubygem-kafo_wizards-doc</packagereq>
-      <packagereq type="default">rubygem-little-plugger-doc</packagereq>
-      <packagereq type="default">rubygem-logging-doc</packagereq>
-      <packagereq type="default">rubygem-multi_json-doc</packagereq>
-      <packagereq type="default">rubygem-oauth-doc</packagereq>
-      <packagereq type="default">rubygem-passenger-doc</packagereq>
-      <packagereq type="default">rubygem-powerbar-doc</packagereq>
-      <packagereq type="default">rubygem-rb-inotify-doc</packagereq>
-      <packagereq type="default">rubygem-rsec-doc</packagereq>
-      <packagereq type="default">rubygem-rubyipmi-doc</packagereq>
```
Comment (ehelms) [inline comps/comps-foreman-rhel7.xml]: Why is it removing all of these?

File: nodejs-extract-text-webpack-plugin/nodejs-extract-text-webpack-plugin.spec [hunk 4]
```diff
@@ -1,92 +1,85 @@
...
 Source15: http://registry.npmjs.org/co/-/co-4.6.0.tgz
 Source16: http://registry.npmjs.org/jsonify/-/jsonify-0.0.0.tgz
-Source17: extract-text-webpack-plugin-2.1.2-registry.npmjs.org.tgz
+Source17: extract-text-webpack-plugin-3.0.0-registry.npmjs.org.tgz
 BuildRequires: nodejs-packaging
-BuildRequires: npm
 BuildArch:  noarch
 ExclusiveArch: %{nodejs_arches} noarch
 
-Provides: npm(%{npm_name}) = %{version}
```
Comment (ehelms) [inline nodejs-extract-text-webpack-plugin/nodejs-extract-text-webpack-plugin.spec]: This removal is going to break builds.

File: nodejs-extract-text-webpack-plugin/nodejs-extract-text-webpack-plugin.spec [hunk 5]
```diff
@@ -1,92 +1,85 @@
...
-%{nodejs_symlink_deps} --check
-#$CHECK
-%endif
+%clean
+rm -rf %{buildroot} %{npm_cache_dir}
 
 %files
 %{nodejs_sitelib}/%{npm_name}
 
+%doc LICENSE
```
Comment (ehelms) [inline nodejs-extract-text-webpack-plugin/nodejs-extract-text-webpack-plugin.spec]: This is likely going to fail from my experiences.

File: nodejs-extract-text-webpack-plugin/nodejs-bundle-extract-text-webpack-plugin.spec (hunk 1 above)
Comment (ehelms) [inline nodejs-extract-text-webpack-plugin/nodejs-bundle-extract-text-webpack-plugin.spec]: Nope, it does not, we need that line or things break. I've hit this twice with updating packages using the script that removes this line.

**Comment by dLobatog** [comment]
Not quite sure why, but it seems that the cache isn't being used and it tries to install online:

```
+ npm install --cache-min Infinity --cache /tmp/npm_cache_nodejs-extract-text-webpack-plugin-3.0.0-1.git.17.8c1bd4b.el7 --no-optional --global-style true extract-text-webpack-plugin@3.0.0
npm ERR! fetch failed https://registry.npmjs.org/extract-text-webpack-plugin/-/extract-text-webpack-plugin-3.0.0.tgz
npm WARN retry will retry, error on last attempt: Error: getaddrinfo ENOTFOUND registry.npmjs.org registry.npmjs.org:443
```

I specifically created this cache for npm 3.10.10 as Koji is using, but no dice. I'll keep on investigating

**Comment by dLobatog** [comment]
Gah, this is the reason why it didn't work - `npm_cache_dir` wasn't being sent properly to `npm install`. I have reverted to the old version of just doing `mkdir npm_cache` in this PR, but a new fix will follow up in npm2rpm.

**Comment by ehelms** [comment]
Are all of the comps/ changes included correct? They seem to modifying packages unrelated to this PR.

**Comment by dLobatog** [comment]
@ehelms Yeah, `comps_doc.sh` automatically detects the packages that have no `%package doc`. All of these comps that have been removed don't have that, therefore it makes sense to remove them (as there's no doc package to build at all). 

Similarly `get-process-mem` does have `%package doc` so it got added.

**Pull request merged**